### ISSUE TITLE
core/lpm_arch: Implement peripheral shutdown and startup

### ISSUE THREAD

**Issue by DipSwitch** [RIOT-OS/RIOT#4281]
It would be nice to have a hook functions from the `_idle_thread` so you can enable / disable peripherals and/or devices before going to sleep. This you would want to overload on the board or application.

Thoughts about this idea?

**Comment by kaspar030** [comment]
Hmm, maybe for the deep-sleep states, otherwise, whenever a thread sleeps (so the scheduler switches to idle), the hooks would need to be executed.

**Comment by punchcard60** [comment]
Having the ability for "applications" and device drivers to power devices on and off when needed is an integral part of power management. I would like to propose that this issue be combined with #2927 as part of Active Power Management. [This wiki page](https://github.com/RIOT-OS/RIOT/wiki/Active-Power-Management-Task-Force) has more information.

**Comment by lebrush** [comment]
@haukepetersen @kaspar030 is this outdated after the PM implementation?

**Comment by kaspar030** [comment]
The concept of the new power management would be that the applications themselves turn off devices when they don't need them anymore, which in turn would make the device unblock sleep modes, which then would make the idle thread select the deepest possible sleep mode.

Would that serve your need?

**Comment by lebrush** [comment]
Mine yes :-) @DipSwitch ?

**Comment by PeterKietzmann** [comment]
I guess it's outdated. Will close, @DipSwitch please reopen if I'm mistaken.
